lib/iris/tests/unit/fileformats/pp_load_rules/test__convert_time_coords.py

```diff
@@ -47,7 +47,9 @@ class TestLBTIMx0x_SingleTimepoint(TestField):
     def _check_timepoint(self, lbcode, expect_match=True):
         lbtim = _lbtim(ib=0, ic=1)
         t1 = nc_datetime(1970, 1, 1, hour=6, minute=0, second=0)
-        t2 = nc_datetime(0, 0, 0)  # not used in result
+        t2 = nc_datetime(
+            0, 0, 0, calendar=None, has_year_zero=True
```

Comment:
Confusingly, you can create a `cftime.datetime(0, 0, 0, calendar=None)` with `has_year_zero` set to either `True` or `False`.

This makes no difference until you come to compare e.g.,
```python
from cftime import datetime

>>> a = datetime(0, 0, 0, calendar=None)
>>> b = datetime(0, 0, 0, calendar=None, has_year_zero=True)
>>> a == a
True
>>> b == b
True
>>> a == b
TypeError: cannot compare cftime.datetime(0, 0, 0, 0, 0, 0, 0, calendar='', has_year_zero=False) and cftime.datetime(0, 0, 0, 0, 0, 0, 0, calendar='', has_year_zero=True)
```
Just for consistency, I'm going to adopt explicitly setting `has_year_zero=True` in this case, although it would be nice for `cftime` to just handle this... in fact, I might raise an issue to see if the devs complain for a good reason or adopt it as a use case :+1:

Reply:
@trexfeathers For the `nimrod_load_rules`, in both instances where a `cftime.datetime` object is created there is surrounding logic to ensure that the `year` used is non-zero. So I think this is okay as it stands, right? 

Just check in case I've missed something

Reply:
The month and day could still be `0` though.

Reply:
Also, see https://github.com/Unidata/cftime/issues/248

Reply:
> The month and day could still be `0` though.

Ahhh sorry, I see what you mean...

My take on this was that the `nimrod` use case is slightly different from the `pp` use case, in that for `nimrod` the `calendar` is explicitly nailed down to being `gregorian`, see https://github.com/SciTools/iris/blob/main/lib/iris/fileformats/nimrod_load_rules.py#L27

They are not ever expecting to see a zero `year`, nor a zero `month` or `day`, as it's indeed an invalid date for that `calendar`. So I think it's more likely that the `nimrod` datetime data is genuinely invalid in this case and an exception should be raised; and it will by `cftime`. 

I think that if we instead returned a valid `cftime.datetime` with `calendar=None`, then that might be more dangerous/unexpected for `nimrod` users.

Reply:
Nice catch, I hadn't spotted that. Totally agree with you.